airgun/entities/task.py

```diff
@@ -20,6 +22,35 @@ def read(self, entity_name, widget_names=None):
         view = self.navigate_to(self, 'Details', entity_name=entity_name)
         return view.read(widget_names=widget_names)
 
+    def set_chart_filter(self, chart_name, index=None):
+        """Remove filter from searchbox ans set filter from specific chart
+
+        :param index: index in 'StoppedChart' table,
+            dict with 'row' number and 'focus' as column name
+        """
+        view = self.navigate_to(self, 'All')
+        chart = getattr(view, chart_name)
+        view.searchbox.clear()
+        if chart_name == 'StoppedChart' and index:
+            chart.table[index['row']][index['focus']].click()
+        else:
+            chart.name.click()
+
+    def read_widget(self, widget_name):
+        """Read specific widget"""
```

Comment:
I feel this goes against the framework, which is built around the idea that tests shouldn't care too much about implementation details behind actions.

Perhaps you could extend `read_all` method to take another argument `widget_names`, and then on test side you would do:
```python
search_term = session.task.read_all('searchbox')
```
? This would be similar to what `read()` does.